handlers/syslog.go

```diff
@@ -17,27 +17,19 @@ var (
 )
 
 type SyslogHandler struct {
-	producer *rabbitmq.Producer
+	producer rabbitmq.Producer
 	debug    bool
 	token    string
 }
 
-func NewSyslogHandler(token string) (*SyslogHandler, error) {
-	var err error
+func NewSyslogHandler(token string, producer rabbitmq.Producer) (*SyslogHandler, error) {
```

Comment:
might be my ignorance, but isn't this a copy of the value. Shouldn't it be by reference?

Reply:
which param are you referring to? Go is always pass by value (even pointers)!

Reply:
With the updated code the producer is an interface